cscs-checks/libraries/math/dgemm.py

```diff
@@ -2,49 +2,87 @@
 import reframe.utility.sanity as sn
 
 
+@rfm.required_version('>=2.14')
+@rfm.simple_test
 class DGEMMTest(rfm.RegressionTest):
     def __init__(self):
         super().__init__()
         self.descr = 'DGEMM performance test'
         self.sourcepath = 'dgemm.c'
-        self.executable_opts = ['5000', '5000', '5000']
-        self.sanity_patterns = sn.assert_found(
-            r'Time for \d+ DGEMM operations', self.stdout)
-        self.maintainers = ['AJ']
-        self.tags = {'production'}
 
+        self.sanity_patterns = self.eval_sanity()
+        # the perf patterns are automaticaly generated inside sanity
+        self.perf_patterns = {}
 
-# FIXME: This test is obsolete; it is kept only for reference.
-@rfm.required_version('>=2.14')
-@rfm.simple_test
-class DGEMMTestMonch(DGEMMTest):
-    def __init__(self):
-        super().__init__()
-        self.tags = {'monch_acceptance'}
-        self.valid_systems = ['monch:compute']
+        self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc']
         self.valid_prog_environs = ['PrgEnv-gnu']
-        self.num_tasks = 1
+
+        self.num_tasks = 0
         self.num_tasks_per_node = 1
         self.num_tasks_per_core = 1
-        self.num_cpus_per_task = 20
-        self.num_tasks_per_socket = 10
+        self.num_tasks_per_socket = 1
         self.use_multithreading = False
-        self.variables = {
-            'OMP_NUM_THREADS': str(self.num_cpus_per_task),
-            'MV2_ENABLE_AFFINITY': '0'
-        }
+
         self.build_system = 'SingleSource'
-        self.build_system.cflags = ['-O3', '-I$EBROOTOPENBLAS/include']
-        self.build_system.ldflags = ['-L$EBROOTOPENBLAS/lib', '-lopenblas',
-                                     '-lpthread', '-lgfortran']
-        self.perf_patterns = {
-            'perf': sn.max(
-                sn.extractall(r'Run\s\d\s+:\s+(?P<gflops>\S+)\s\S+',
-                              self.stdout, "gflops", float)
-            )
+        self.build_system.cflags = ['-O3']
+
+        self.sys_reference = {
+            'daint:gpu': (460, -0.1, None),
+            'daint:mc': (460, -0.1, None),
+            'dom:gpu': (460, -0.1, None),
+            'dom:mc': (460, -0.1, None),
+            'monch:compute': (350, -0.1, None),
         }
-        self.reference = {
-            'monch:compute': {
-                'perf': (350, -0.1, None)
-            }
+
+        self.maintainers = ['AJ', 'VH', 'VK']
+        self.tags = {'production'}
```

Comment:
Please remove this tag. It causes CI to fail. We should put sth else, perhaps `flex` or `flexible`.